File: src/main/java/cn/zicla/blog/rest/base/BaseEntity.java
package cn.zicla.blog.rest.base;

import cn.zicla.blog.util.DateUtil;
import cn.zicla.blog.util.JsonUtil;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;
import java.io.Serializable;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

@Data
@MappedSuperclass
public class BaseEntity extends Base implements Serializable {

    public final static String PREFIX = "blog_";

    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "org.hibernate.id.UUIDGenerator")
    private String uuid;

    public long sort = System.currentTimeMillis();

    //创建时间
    @JsonFormat(pattern = DateUtil.DEFAULT_FORMAT)
    @CreationTimestamp
    public Date createTime = new Date();

    //更新时间
    @JsonFormat(pattern = DateUtil.DEFAULT_FORMAT)
    @UpdateTimestamp
    public Date updateTime = new Date();


    @JsonIgnore
    public Boolean deleted = false;



}

File: src/main/resources/static/js/home/article/BaseEntity.js
//定义Base实体。
function BaseEntity() {
    //继承Base的属性
    Base.call(this, arguments);

    this.uuid = null
    this.sort = null
    this.createTime = null
    this.modifyTime = null
    this.deleted = false

}

//继承Base的方法
BaseEntity.prototype = new Base();
BaseEntity.prototype.constructor = BaseEntity;

BaseEntity.prototype.render = function (obj) {
    if (obj) {

        Base.prototype.render.call(this, obj);
        this.createTime = str2Date(obj.createTime)
        this.modifyTime = str2Date(obj.modifyTime)
    }
};
